source/SH3/arc/vfile.cpp

```diff
@@ -55,15 +60,22 @@ void sh3_arc_vfile::Seek(long pos, std::ios_base::seekdir origin)
 {
     if(origin == std::ios_base::beg)
     {
-        fpos = pos;
+        fpos = static_cast<decltype(fpos)>(pos);
     }
-    else if(std::ios_base::cur)
+    else if(origin == std::ios_base::cur)
     {
-        fpos += pos;
```

Comment:
The comment below gives a rough explanation why it's fine. Feel free to verify that it's true though.